src/Summoner/License.hs

```diff
@@ -1,51 +1,81 @@
+{-# LANGUAGE LambdaCase #-}
+
 module Summoner.License
-       ( License (..)
-       , customizeLicense
-       , licenseNames
-       , githubLicenseQueryNames
-       ) where
+    ( LicenseName(..)
+    , License(..)
+    , cabalLicenseName
+    , customizeLicense
+    , githubLicenseQueryNames
+    , parseLicense
+    )
+where
 
 import Relude
+import Relude.Extra.Enum (inverseMap)
 
 import Data.Aeson (FromJSON (..), withObject, (.:))
-
 import qualified Data.Text as T
 
 ----------------------------------------------------------------------------
 -- License
 ----------------------------------------------------------------------------
 
-licenseNames :: [License]
-licenseNames = map fst githubLicenseQueryNames
-
-githubLicenseQueryNames :: [(License, Text)]
-githubLicenseQueryNames =
-    [ ("MIT",        "mit")
-    , ("BSD2",       "bsd-2-clause")
-    , ("BSD3",       "bsd-3-clause")
-    , ("GPL-2",      "gpl-2.0")
-    , ("GPL-3",      "gpl-3.0")
-    , ("LGPL-2.1",   "lgpl-2.1")
-    , ("LGPL-3",     "lgpl-3.0")
-    , ("AGPL-3",     "agpl-3.0")
-    , ("Apache-2.0", "apache-2.0")
-    , ("MPL-2.0",    "mpl-2.0")
-    ]
+data LicenseName
+    = MIT
+    | BSD2
+    | BSD3
+    | GPL2
+    | GPL3
+    | LGPL21
+    | LGPL3
+    | AGPL3
+    | Apache20
+    | MPL20
+    deriving (Eq, Ord, Show, Enum, Bounded, Generic)
+
+githubLicenseQueryNames :: LicenseName -> Text
+githubLicenseQueryNames = \case
+    MIT      -> "mit"
+    BSD2     -> "bsd-2-clause"
+    BSD3     -> "bsd-3-clause"
+    GPL2     -> "gpl-2.0"
+    GPL3     -> "gpl-3.0"
+    LGPL21   -> "lgpl-2.1"
+    LGPL3    -> "lgpl-3.0"
+    AGPL3    -> "agpl-3.0"
+    Apache20 -> "apache-2.0"
+    MPL20    -> "mpl-2.0"
+
+cabalLicenseName :: LicenseName -> Text
+cabalLicenseName = \case
```

Comment:
It's better to make this `show` function instead of useless derived instance :slightly_smiling_face:

Reply:
Yeah, I agree. Probably it's even better to have useful `Show` instance and don't have `cabalLicenseName` as a separate function.